src/xb-silo-export.c

```diff
@@ -45,47 +45,54 @@ xb_silo_export_node (XbSilo *self, XbSiloExportHelper *helper, XbSiloNode *sn, G
 		g_string_append_printf (helper->xml, " %s=\"%s\"", key, val);
 	}
 
-	/* finish the opening tag and add any text if it exists */
-	if (sn->text != XB_SILO_UNSET) {
-		g_autofree gchar *text = xb_string_xml_escape (xb_silo_from_strtab (self, sn->text));
-		g_string_append (helper->xml, ">");
-		g_string_append (helper->xml, text);
+	/* collapse open/close tags together if no text or children */
+	if ((helper->flags & XB_NODE_EXPORT_FLAG_COLLAPSE_IF_NO_CHILDREN) &&
+	    (sn->text == XB_SILO_UNSET) &&
+	    (xb_silo_node_get_child (self, sn) == NULL)) {
+		g_string_append (helper->xml, "/>");
```

Comment:
I think you want a space here; most XML i've seen has the extra char.

Reply:
Done